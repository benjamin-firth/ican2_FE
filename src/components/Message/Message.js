import React from 'react';
import './Message.scss';

const Message = () => {

  return (
    <section>
      <p>This is a message.</p>
    </section>
  );
}

export default Message;
