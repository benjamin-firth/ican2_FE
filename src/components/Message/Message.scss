.message {
  display: flex;
  margin-top: 1vw;
  
  img {
    width: 6vw;
    height: 6vw;
    border-radius: 50%;
  }

  p {
    background-color: white;
    width: 30vw;
    margin: 1vw;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}

.sender {
  justify-content: flex-end;
}

.user {
  justify-content: start;
}
