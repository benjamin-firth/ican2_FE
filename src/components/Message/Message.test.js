import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message';

const mockState = {
  currentUser: {
    id: 2,
    profile: {
      image: 'url'
    }
  },
  messages: {otherMessenger: {
      id: 3
    }
  }
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState
}));

describe('Message', () => {
  let wrapper;
  const mockMessage = {sender: 2, recipient: 1, body: 'hello'};

  beforeEach(() => {
    wrapper = shallow( <Message message={mockMessage}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
