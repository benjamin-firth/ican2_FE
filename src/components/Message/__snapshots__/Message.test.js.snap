// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match the snapshot 1`] = `
<section
  className="message sender"
>
  <p>
    hello
  </p>
  <img
    src="url"
  />
</section>
`;
